File: ws/bohorquez/U2/WS15Inheritans/ZooApp/src/main/java/ec/edu/espe/model/Animal.java
package ec.edu.espe.model;
import java.util.ArrayList;
import java.util.Date;

/**
 *
 * @author Camila Bohorquez
 */
public class Animal {
    private int id;
    private String name;
    private Date bornOnDate;
    private float weight;
    private char sex;
    private ArrayList<Vaccine> vaccines;

    public Animal(int id, String name, Date bornOnDate, float weight, char sex, ArrayList<Vaccine> vaccines) {
        this.id = id;
        this.name = name;
        this.bornOnDate = bornOnDate;
        this.weight = weight;
        this.sex = sex;
        this.vaccines = vaccines;
    }

    public void feed(Food food) {
        System.out.println("Feeding the animal " + name + " with " + food);
    }

    public void register() {
        System.out.println("Registering the animal " + name);
    }

    public void assignCage(Cage cage) {
        System.out.println("Assigning the animal " + name + " to cage " + cage);
        cage.getAnimals().add(this);
    }

    public void provideVaccination(Vaccine vaccine) {
        System.out.println("Providing vaccination to the animal " + name + " with " + vaccine);
        this.vaccines.add(vaccine);
    }

    @Override
    public String toString() {
        return "Animal{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", bornOnDate=" + bornOnDate +
                ", weight=" + weight +
                ", sex=" + sex +
                ", vaccines=" + vaccines +
                '}';
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Date getBornOnDate() {
        return bornOnDate;
    }

    public void setBornOnDate(Date bornOnDate) {
        this.bornOnDate = bornOnDate;
    }

    public float getWeight() {
        return weight;
    }

    public void setWeight(float weight) {
        this.weight = weight;
    }

    public char getSex() {
        return sex;
    }

    public void setSex(char sex) {
        this.sex = sex;
    }

    public ArrayList<Vaccine> getVaccines() {
        return vaccines;
    }

    public void setVaccines(ArrayList<Vaccine> vaccines) {
        this.vaccines = vaccines;
    }
}

File: hw/quiroga/U2/Hw18Exceptions/Inheritance_JS/javascript/ec/edu/espe/zoo/model/Animal.mjs
export class Animal {
    _id;
    #name;
    #bornOnDate;
    #weight;
    #sex;
    #vaccines;

    constructor(id, name, bornOnDate, weight, sex, vaccines) {
        this._id = id;
        this.#name = name;
        this.#bornOnDate = bornOnDate;
        this.#weight = weight;
        this.#sex = sex;
        this.#vaccines = vaccines;
    }

    feed(food) {
        console.log("Feeding the animal with -->" + food);
    }

    register() {
        console.log("Registering the animal --> " + this.#name);
    }

    assignCage(cage) {
        console.log("Assigning an animal -- >" + this.#name + " in cage -->" + cage);
        cage.getAnimals().push(this);
    }

    provideVaccination(vaccine) {
        console.log("Providing vaccine --> 0" + vaccine + "<-- to " + this.#name);
        this.#vaccines.push(vaccine);
    }

    toString() {
        return `Animal {id: ${this._id}, name: ${this.#name}, bornOnDate: ${this.#bornOnDate}, weight: ${this.#weight}, sex: ${this.#sex}, vaccines: ${this.#vaccines.join(', ')}}`;
    }

    getId() {
        return this._id;
    }

    setId(id) {
        this._id = id;
    }

    getName() {
        return this.#name;
    }

    setName(name) {
        this.#name = name;
    }

    getBornOnDate() {
        return this.#bornOnDate;
    }

    setBornOnDate(bornOnDate) {
        this.#bornOnDate = bornOnDate;
    }

    getWeight() {
        return this.#weight;
    }

    setWeight(weight) {
        this.#weight = weight;
    }

    getSex() {
        return this.#sex;
    }

    setSex(sex) {
        this.#sex = sex;
    }

    getVaccines() {
        return this.#vaccines;
    }

    setVaccines(vaccines) {
        this.#vaccines = vaccines;
    }
}
